src/common.h

```diff
@@ -16,15 +16,111 @@
 #define COMMON_H_
 
 #include <cstring>
+#include <iostream>
 #include <memory>
 #include <string>
 #include <utility>
 #include <vector>
 
-#include <glog/logging.h>
 #include "src/deps/basic_types.h"
 #include "src/deps/canonical_errors.h"
 #include "src/deps/status.h"
 #include "src/deps/status_macros.h"
 
+#if defined(_WIN32) && !defined(__CYGWIN__)
+#define OS_WIN
+#else
+#define OS_UNIX
+#endif
+
+namespace sentencepiece {
+namespace error {
+
+void Abort();
+void Exit(int code);
+void SetTestCounter(int c);
+void ResetTestMode();
+bool GetTestCounter();
+
+class Die {
+ public:
+  explicit Die(bool die) : die_(die) {}
+  ~Die() {
+    std::cerr << std::endl;
+    if (die_) {
+      Abort();
+    }
+  }
+  int operator&(std::ostream &) { return 0; }
+
+ private:
+  bool die_;
+};
+}  // namespace error
+
+namespace logging {
+enum LogSeverity {
+  LOG_INFO = 0,
+  LOG_WARNING = 1,
+  LOG_ERROR = 2,
+  LOG_FATAL = 3,
+  LOG_SEVERITY_SIZE = 4,
+};
+
+inline const char *BaseName(const char *path) {
+#ifdef OS_WIN
+  const char *p = strrchr(path, '\\');
+#else
+  const char *p = strrchr(path, '/');
+#endif
+  if (p == nullptr) return path;
+  return p + 1;
+}
+}  // namespace logging
+}  // namespace sentencepiece
+
+#define LOG(severity)                                                     \
+  ::sentencepiece::error::Die(::sentencepiece::logging::LOG_##severity >= \
+                              ::sentencepiece::logging::LOG_FATAL) &      \
+      std::cerr << ::sentencepiece::logging::BaseName(__FILE__) << "("    \
+                << __LINE__ << ") "                                       \
+                << "LOG(" << #severity << ") "
+
+#define CHECK(condition)                                                      \
+  (condition) ? 0                                                             \
+              : ::sentencepiece::error::Die(true) &                           \
+                    std::cerr << ::sentencepiece::logging::BaseName(__FILE__) \
+                              << "(" << __LINE__ << ") [" << #condition       \
+                              << "] "
```

Comment:
Here too. Will the program exit before the message is printed?

Reply:
Same as above.